volatility3/framework/plugins/linux/iomem.py

```diff
@@ -0,0 +1,156 @@
+# This file is Copyright 2023 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+import logging
+from typing import List
+
+from volatility3.framework import renderers, interfaces, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import format_hints
+
+vollog = logging.getLogger(__name__)
+
+
+class IOMem(interfaces.plugins.PluginInterface):
+    """Generates an output similar to /proc/iomem on a running system."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            )
+        ]
+
+    @classmethod
+    def parse_resource(
+        cls,
+        context: interfaces.context.ContextInterface,
+        vmlinux_module_name: str,
+        resource_offset: int,
+        seen: set = set(),
+        depth: int = 0,
+    ):
+        """Recursively parse from a root resource to find details about all related resources.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            vmlinux_module_name: The name of the kernel module on which to operate
+            resource_offset: The offset to the resouce to be parsed
+            seen: The set of resource offsets that have already been parsed
+            depth: How deep into the resource structure we are
+
+        Yields:
+            Each row of output
+        """
+        vmlinux = context.modules[vmlinux_module_name]
+
+        # create the resource object with protection against memory smear
+        try:
+            resource = vmlinux.object("resource", resource_offset)
+        except exceptions.InvalidAddressException:
+            vollog.warning(
+                f"Unable to create resource object at {resource_offset:#x}. This resource, "
+                "its sibling, and any of it's childern and will be missing from the output."
+            )
+            return None
+
+        # extract the information required for this resource
+        start = format_hints.Hex(resource.start)
```

Comment:
Generally it makes it slightly easier to read and reason about if the format_hint is only applied at the moment of use (if, during the yield statement).  Otherwise someone unfamiliar with the code may wonder why it needs putting through this function/class and waste time because we did the action far away from where it was needed.